File: lib/core/providers/contracts/car_provider.dart
import '../../../core/domain/models/offers/car_offer.dart';

abstract interface class CarProvider {
  String get providerId;
  String get providerName;

  Future<List<CarOffer>> searchCars({
    required String pickupLocation,
    required DateTime pickupTime,
    required DateTime dropoffTime,
  });
}

File: backend/src/common/providers/car.provider.ts
import { ProviderResult } from './provider.result';

export interface CarProvider {
  providerId: string;
  providerName: string;
  searchCars(params: {
    pickupLocation: string;
    pickupTime: Date;
    dropoffTime: Date;
  }): Promise<ProviderResult<any[]>>;
}
